server/src/main/java/org/elasticsearch/search/aggregations/AggregatorBase.java

```diff
@@ -105,6 +108,22 @@ public ScoreMode scoreMode() {
         addRequestCircuitBreakerBytes(DEFAULT_WEIGHT);
     }
 
+    /**
+     * Returns a converter for point values if early termination is applicable to
+     * the context or <code>null</code> otherwise.
+     *
+     * @param config The config for the values source metric.
+     */
+    public Function<byte[], Number> getPointReaderOrNull(ValuesSourceConfig config) {
```

Comment:
Do you think now'd be a good time to change the name? I always thought something like `pointReaderIfSafe` or something is a little more clear about *why* you might get null back.

And it isn't so much "if early termination is applicable" as "if there are no filters and the underlying index is in the same order as the values produced by the config"

Reply:
Could you make the method `final`?

Reply:
> the underlying index is in the same order as the values produced by the config

Out of curiosity, what in the current logic looks at index ordering?